test/test_linalg.py

```diff
@@ -8006,6 +8006,26 @@ def test_tensordot(self, device):
         an = torch.from_numpy(np.tensordot(np.zeros((), dtype=np.float32), np.zeros((), dtype=np.float32), 0))
         self.assertEqual(a, an)
 
+    def test_linalg_trace(self, device):
+        shapes = [(1, 1), (10, 1), (1, 10), (2, 3, 4), (4, 3, 2)]
+        offsets = [0, 1, -1, 2, -2, 10, -10]
+
+        for shape in shapes:
+            size = np.prod(shape)
+            x = torch.arange(size, device=device).reshape(*shape)
+            y = torch.linalg.trace(x)
+            xn = np.array(x.cpu().numpy()).reshape(shape)
+            yn = np.trace(xn, axis1=-2, axis2=-1)
```

Comment:
This might work without all the explicit castings by simply doing

```suggestion
            yn = np.trace(x.cpu(), axis1=-2, axis2=-1)
```
Same below.